### DISCUSSION TITLE
[Pitch] Support for .sqlite based formats

### DISCUSSION THREAD

**Discussion by jspaezp** [CompOmics/psm_utils#58] [Ideas]
Hello there!

I was wondering if there is any desire to add support for sqlite-based spectral library formats?
In particular I am thinking of skyline/bibliospec's `.blib` and Encyclopedia's `.elib/dlib`.
LMK what you think.

[https://raw.githubusercontent.com/ProteoWizard/pwiz/master/pwiz_tools/BiblioSpec/tests/reference/tables.check](https://raw.githubusercontent.com/ProteoWizard/pwiz/master/pwiz_tools/BiblioSpec/tests/reference/tables.check)

I have some experience packing my own for those and could make a couple of PRs adding them (if there is desire for them and we decide on the scope on how to deal with the 'rough edges').

The main issue I see here would be that creating those files requires having actual spectra and not just the PSM information, so it cannot be generated uniquely from a .pin for instance (it requires some raw data with it). But since pyteomics is also a dependency, we could read the companion .mzML to get them.

LMK what you think
Best,
Sebastian

**Comment by RalfG** [comment]
Hi Sebastian!

Reading for PSM information makes sense, but writing PSM files + spectrum files to a library is an interesting proposition. I think it can perfectly fit within the current framework, although it might include adding functionality that I originally did not envision as part of this project. For instance, the spectra will have to be annotated (for which we could use spectrum_utils). 

Perhaps a better suited project for such functionality would be the [mzSpecLib Python package](https://github.com/HUPO-PSI/mzSpecLib/tree/master/implementations/python/), mostly built by @mobiusklein. This package can already interconvert between various spectral library formats, including skyline/bibliospec `.blib`. Then I could see an implementation where one could convert PSM files (read with psm_utils) and spectrum files (read with pyteomics, parsed and annotated by spectrum_utils) to spectral libraries to be written by the mzSpecLib package.

These are just some of my current thoughts. I'm definitely open to adding more functionality to psm_utils (and all contributions are welcome 😄). My only concern here would be to keep the package (and the wider ecosystem) modular and not too convoluted.

Best,
Ralf
